dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -1971,6 +1971,19 @@ void DeltaMergeStore::segmentMerge(DMContext & dm_context, const SegmentPtr & le
         right->info(),
         dm_context.min_version);
 
+    /// This segment may contain some rows that not belong to this segment range which is left by previous split operation.
+    /// And only saved data in this segment will be filtered by the segment range in the merge process,
+    /// unsaved data will be directly copied to the new segment.
+    /// So we flush here to make sure that all potential data left by previous split operation is saved.
+    while (!left->flushCache(dm_context))
+    {
+        // keep flush until success
+    }
+    while (!right->flushCache(dm_context))
+    {
+        // keep flush until success
+    }
```

Comment:
If the `left`/`right` segment gets abandoned by another thread, will it run into an inf loop? Should we check `left->hasAbandoned`/`right->hasAbandoned` in the while loop?

https://github.com/pingcap/tiflash/blob/2e8d324e76a2c5a96c83fdab8e7e08f3a5424d61/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp#L142-L207

Reply:
+1. Beet to check validation of segment instant. If is invalid then simply give up.

Reply:
Fixed.